src/main/kotlin/ar/edu/unahur/obj2/vendedores/CentroDistribucion.kt

```diff
@@ -0,0 +1,25 @@
+package ar.edu.unahur.obj2.vendedores
+
+abstract class CentroDistribucion {
+    val vendedores = mutableListOf<Vendedor>()
+    val genericos = mutableListOf<Vendedor>()
+
+    fun agregarVendedoresNuevoYGenericoOError(vendedor: Vendedor) {
+        if (vendedores.contains(vendedor)) {
+            throw Exception("el vendedor ya esta en el centro")
+        }
+        else {
+            this.agregarVendedor(vendedor)
+            this.agregarSiEsGenerico(vendedor)
+        }
+    }
+    fun agregarVendedor(vendedor: Vendedor) { vendedores.add(vendedor) }
+    fun agregarSiEsGenerico(vendedor : Vendedor) {
+        if(vendedor.esGenerico()) { genericos.add(vendedor) }
+    }
```

Comment:
Esto es redundate, si un vendedor es generico lo estarías guardando 2 veces, en diferentes listas, una solución mejor sería tener sólo una  lista de vendedores y filtrar los que son genericos.
```suggestion
    fun vendedoresGenericos() = vendedores.filter { it -> it.esGenerico() }
    }
```